feat: Support `coupon`s for Revenue Analytics

rafaeelaudibert opened the pull request:
We can now filter/breakdown both by the coupon in a specific revenue event, but also the initial coupon used by a person on their first invoice.

rafaeelaudibert committed:
feat: Support tooltips on `LemonInputSelect`

This could be achieved with `labelComponent` but it's much easier to achieve with a custom prop

rafaeelaudibert committed:
feat: Support `coupon`s for Revenue Analytics

We can now filter/breakdown both by the coupon in a specific revenue event, but also the initial coupon used by a person on their first invoice.

rafaeelaudibert committed:
refactor: Sort alphabetically

rafaeelaudibert commented on products/revenue_analytics/guides/new_option_for_filters_and_group_by.md:
````diff
@@ -140,7 +140,6 @@ Add the new case inside `append_group_by()`
 ```python
 def _get_subquery(self) -> ast.SelectQuery | None:
     if group_by == RevenueAnalyticsInsightsQueryGroupBy.COUNTRY:
-        subquery = customer_subquery
         join_to = RevenueAnalyticsCustomerView
         field_name = "country"
````
Intended

rafaeelaudibert commented on products/revenue_analytics/frontend/RevenueAnalyticsFilters.tsx:
```diff
@@ -106,12 +107,17 @@ export const RevenueAnalyticsFilters = (): JSX.Element => {
 const BREAKDOWN_BY_MAPPING: Record<RevenueAnalyticsInsightsQueryGroupBy, string> = {
     [RevenueAnalyticsInsightsQueryGroupBy.COHORT]: 'Cohort',
     [RevenueAnalyticsInsightsQueryGroupBy.COUNTRY]: 'Country',
+    [RevenueAnalyticsInsightsQueryGroupBy.COUPON]: 'Coupon',
+    [RevenueAnalyticsInsightsQueryGroupBy.COUPON_ID]: 'Coupon ID',
+    [RevenueAnalyticsInsightsQueryGroupBy.INITIAL_COUPON]: 'Initial coupon',
+    [RevenueAnalyticsInsightsQueryGroupBy.INITIAL_COUPON_ID]: 'Initial coupon ID',
     [RevenueAnalyticsInsightsQueryGroupBy.PRODUCT]: 'Product',
 }
```
This is alphabetized

jabahamondes reviewed (approved):
Really nice PR to review EOD 🚀

rafaeelaudibert committed:
refactor: Simplify code required to be changed when adding new field

We can make this much simpler by extracting some of the logic to some different methods

github-actions[bot] committed:
Update query snapshots

rafaeelaudibert committed:
refactor: Fix types

rafaeelaudibert committed:
refactor: Extract Python code quality check to separate file

This avoids the need to run this twice: dagster AND backend

robbie-c reviewed (approved):
IMO this could have been 2 separate PRs (ci changes and everything else) but LGTM

rafaeelaudibert committed:
fix: Disable settings snapshots for now
